virttest/qemu_vm.py

```diff
@@ -455,9 +455,15 @@ def add_qmp_monitor(devices, monitor_name, filename):
             return cmd
 
         def add_serial(devices, name, filename):
+            # Arm lists "isa-serial" as supported but can't use it,
+            # fallback to "-serial"
             if (not devices.has_option("chardev") or
-                    not any(devices.has_device(dev)
-                            for dev in ("isa-serial", "sclpconsole", "spapr-vty"))):
+                    'arm' in params.get("machine_type", "")):
+                return " -serial unix:'%s',server,nowait" % filename
```

Comment:
Since the string has been referenced twice, I'd suggest to assign it to a var, like:
```python
legacy_cmd = " -serial unix:'%s',server,nowait" % filename
if ...:
    return legacy_cmd
for ...:
    ...
else:
    return legacy_cmd
```

Reply:
sure